Automattic/wp-calypso
# Date Range Component

## Changes proposed in this Pull Request

Add a new `DateRange` component, extracted from changes originally proposed in [`25762`](https://github.com/Automattic/wp-calypso/pull/25762).

## Testing instructions

This is a new component so please follow the Component Checklist README.

Also:
* View and test the example at http://calypso.localhost:3000/devdocs/design/date-range
* Run the tests `npm run test-client:watch client/components/date-range/test`
* Check the existing DateRange component in the `Activity` section of Calypso - ref: https://github.com/Automattic/wp-calypso/pull/29938/commits/22fcf7f367047efd0d58ebc30f58066e7224bfa4

## Todo

- [x] ~~Switch calendar to month when start/end input receives `focus`~~
- [x] ~~Match styling of range to the range in the Activity log~~
- [x] ~~Test iOS regression re: touch event handlers (see https://github.com/Automattic/wp-calypso/commit/406a5f6226761b5ef3ff17f35f5d42956880ea6e)~~
- [x] ~~Resolve timezone issues highlighted by [this feedback](https://github.com/Automattic/wp-calypso/pull/29938#issuecomment-453569311). Likely need to convert all tests to use `ISO` format for assertions to ensure consistent results~~
- [x] ~~Update color vars as per `p4TIVU-9c1-p2`~~
- [x] ~~Address @mattsherman CR points~~
- [x] ~~Document render props~~
- [x] ~~DRY up date range clamping validation code~~
- [x] ~~Clamping of start/end date props to ensure they conform to `first|lastSelectableDate` prop ranges~~
- [x] ~~https://github.com/Automattic/wp-calypso/pull/25762#pullrequestreview-189859746~~
- [x] ~~https://github.com/Automattic/wp-calypso/pull/25762#pullrequestreview-189774450~~
- [x] ~~https://github.com/Automattic/wp-calypso/pull/25762#discussion_r245709897~~
- [x] ~~https://github.com/Automattic/wp-calypso/pull/29938#discussion_r245840735~~
- [x] ~~https://github.com/Automattic/wp-calypso/pull/29938#discussion_r245838307~~
- [x] ~~Refactor tests~~
